core/Pos/Crypto/SafeSigning.hs

```diff
@@ -76,3 +115,8 @@ withSafeSigner
     -> (SafeSigner -> m a)
     -> m a
 withSafeSigner sk ppGetter action = ppGetter >>= action . SafeSigner sk
+
+-- | We need this to be able to perform signing with unencrypted `SecretKey`s,
+-- where `SafeSigner` is required
+fakeSigner :: SecretKey -> SafeSigner
+fakeSigner = FakeSigner
```

Comment:
Why not use `FakeSigner` constructor directly? Does this function supposed to be changed and turned into smart constructor?